RELEASE_NOTES.md

```diff
@@ -1,5 +1,7 @@
 # Release notes
 
+## New in git-machete 3.16.3
+
```

Comment:
Pls add an entry to release notes (plus I've released 3.16.3 in the meantime + moved this one to 3.16.4)